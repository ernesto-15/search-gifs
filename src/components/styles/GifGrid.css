.card-grid {
  display: grid;
  grid-template-columns: 1fr;
  gap: 2rem;
  width: 100%;
}

.container {
  margin: 35px auto;
}

.container h3 {
  font-size: 2rem;
  text-align: center;
  margin-bottom: 25px;
  color: white;
}

@media (min-width: 768px) {
  .card-grid {
    grid-template-columns: repeat(2, 1fr);
  }

  .container {
    max-width: 1000px;
  }
}

@media (min-width: 1024px) {
  .card-grid {
    grid-template-columns: repeat(3, 1fr);
  }
}
