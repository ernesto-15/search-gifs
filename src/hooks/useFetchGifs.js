import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGifs';

export const useFetchGifs = (category) => {
  const [data, setData] = useState(null);
  const [loading, setLoading] = useState(true);

  useEffect(() => {
    getGifs(category).then((images) => {
      setData([...images]);
      setLoading(false);
    });
  }, [category]);

  return {
    data,
    loading,
  };
};
