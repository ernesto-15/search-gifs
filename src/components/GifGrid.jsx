import React from 'react';
import PropTypes from 'prop-types';
import GifGridItem from './GifGridItem';
import './styles/GifGrid.css';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ({ category }) => {
  const { data, loading } = useFetchGifs(category);

  // if (loading && !data) {
  //   return (
  //     <p className="loading animate__animated animate__heartBeat animate__infinite"></p>
  //   );
  // }

  return (
    <div className="container">
      <h3>{category}</h3>
      {loading && (
        <p className="loading animate__animated animate__heartBeat animate__infinite"></p>
      )}
      <div className="card-grid">
        {data.map((image) => (
          <GifGridItem {...image} key={image.id} />
        ))}
      </div>
    </div>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};

export default GifGrid;
