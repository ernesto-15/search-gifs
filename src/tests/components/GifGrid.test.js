import React from 'react';
import GifGrid from '../../components/GifGrid.jsx';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import { shallow } from 'enzyme';

jest.mock('../../hooks/useFetchGifs');

describe('tests in GifGrid component', () => {
  const category = 'Rick and Morty';

  test('should show component', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show items when images are feteched with useFetchGifs', () => {
    const images = [
      {
        id: 'abc',
        url: 'https://yes/yes.jpg',
        title: 'idk',
      },
      {
        id: 'abcd',
        url: 'https://yes/yes.jpg',
        title: 'idk2',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: images,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBeFalsy();
    expect(wrapper.find('GifGridItem').length).toBe(images.length);
  });
});
