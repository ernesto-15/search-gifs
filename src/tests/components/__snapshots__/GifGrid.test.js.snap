// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests in GifGrid component should show component 1`] = `
<div
  className="container"
>
  <h3>
    Rick and Morty
  </h3>
  <p
    className="loading animate__animated animate__heartBeat animate__infinite"
  />
  <div
    className="card-grid"
  />
</div>
`;

exports[`tests in GifGrid component should show items when images are feteched with useFetchGifs 1`] = `
<div
  className="container"
>
  <h3>
    Rick and Morty
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="idk"
      url="https://yes/yes.jpg"
    />
    <GifGridItem
      id="abcd"
      key="abcd"
      title="idk2"
      url="https://yes/yes.jpg"
    />
  </div>
</div>
`;
